### PR TITLE
Improve debug/info log for RPC request in the server

### PR DESCRIPTION
**Pull request by touilleMan**
new format example:
image

and with log level debug:
image

### CODE REVIEW

File: server/parsec/config.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ class BackendConfig:
     forward_proto_enforce_https: tuple[str, str] | None
     server_addr: ParsecAddr | None
 
+    # If log level is DEBUG, we sometime want to replace some INFO logs by
+    # equivalent ones with more details.
+    log_level_is_debug: bool
```
Comment (vxgmichel) [inline server/parsec/config.py:120]: Why not simply `log_level`?

This would be more flexible than a single boolean.

**Review by vxgmichel — APPROVED** [review]
A single comment, LGTM otherwise :+1:

File: server/parsec/config.py (hunk 1 above)
Comment (touilleMan) [reply server/parsec/config.py:120]: YAGNI kid, you should focus more on KISS :trollface:

File: server/parsec/asgi/rpc.py [hunk 2]
```diff
@@ -428,8 +428,26 @@ async def anonymous_api(raw_organization_id: str, request: Request) -> Response:
...
+    client_ctx.logger.debug(
+        "RPC start",
+        req=req,
+    )
+
     cmd_func = backend.apis[type(req)]
     rep = await cmd_func(client_ctx, req)
+
+    if backend.config.log_level_is_debug:
+        client_ctx.logger.info(
```
Comment (vxgmichel) [inline server/parsec/asgi/rpc.py:440]: Shoudn't this be `logger.debug` instead?
```suggestion
        client_ctx.logger.debug(
```
This is what you did in the `event` component. Or use `logger.info` in the `event` component for consistency.

**Review by vxgmichel — CHANGES_REQUESTED** [review]

File: server/parsec/asgi/rpc.py (hunk 2 above)
Comment (touilleMan) [reply server/parsec/asgi/rpc.py:440]: :+1:

**Review by vxgmichel — APPROVED** [review]
A refactoring suggestion:
```diff
diff --git a/server/parsec/asgi/rpc.py b/server/parsec/asgi/rpc.py
index 043c140da..7c24b2d82 100644
--- a/server/parsec/asgi/rpc.py
+++ b/server/parsec/asgi/rpc.py
@@ -437,17 +437,14 @@ async def anonymous_api(raw_organization_id: str, request: Request) -> Response:
     cmd_func = backend.apis[type(req)]
     rep = await cmd_func(client_ctx, req)
 
-    if backend.config.log_level == LogLevel.DEBUG:
-        client_ctx.logger.info(
-            "RPC done",
-            rep=rep,
-        )
-    else:
-        client_ctx.logger.info(
-            "RPC",
-            cmd=cmd_func._api_info["cmd"],  # type: ignore
-            status=type(rep).__name__,
-        )
+    backend.config.log_level.bind_for_debug(
+        client_ctx.logger,
+        rep=rep,
+    ).info(
+        "RPC",
+        cmd=cmd_func._api_info["cmd"],  # type: ignore
+        status=type(rep).__name__,
+    )
 
     return _rpc_rep(rep, parsed.settled_api_version)
 
diff --git a/server/parsec/config.py b/server/parsec/config.py
index 134faa903..d36b717ee 100644
--- a/server/parsec/config.py
+++ b/server/parsec/config.py
@@ -6,6 +6,8 @@ from dataclasses import dataclass, field
 from enum import Enum
 from typing import TYPE_CHECKING
 
+import structlog
+
 from parsec._parsec import ActiveUsersLimit, ParsecAddr
 
 if TYPE_CHECKING:
@@ -109,6 +111,14 @@ class LogLevel(Enum):
     ERROR = logging.ERROR
     CRITICAL = logging.CRITICAL
 
+    def bind_for_debug(
+        self, logger: structlog.stdlib.BoundLogger, **kwargs: object
+    ) -> structlog.stdlib.BoundLogger:
+        if self == LogLevel.DEBUG:
+            return logger.bind(**kwargs)
+        else:
+            return logger.bind()
+
 
 @dataclass(slots=True)
 class BackendConfig:
```
Related SO question: https://stackoverflow.com/questions/78554551/adapt-context-to-log-level-in-structlog

LGTM otherwise :+1:

File: server/parsec/components/memory/events.py [hunk 3]
```diff
@@ -42,7 +43,20 @@ async def event_bus_factory() -> AsyncIterator[MemoryEventBus]:
...
+                    "Received internal event",
+                    event_=event,
+                )
+            else:
+                logger.info(
+                    "Received internal event",
+                    event_type=event.type,
+                    event_id=event.event_id.hex,
+                    organization=event.organization_id.str,
+                )
```
Comment (vxgmichel) [inline server/parsec/components/memory/events.py:58]: Why not use `info_with_debug_extra` here?

**Review by vxgmichel — APPROVED** [review]
Much better now! A single comment tho

File: server/parsec/config.py [hunk 4]
```diff
@@ -115,6 +125,9 @@ class BackendConfig:
     forward_proto_enforce_https: tuple[str, str] | None
     server_addr: ParsecAddr | None
 
+    # If log level is DEBUG, we sometime want to replace some INFO logs by
+    # equivalent ones with more details.
+    log_level: LogLevel
```
Comment (vxgmichel) [inline server/parsec/config.py:130]: So this is not used anymore right?

File: server/parsec/components/postgresql/events.py [hunk 5]
```diff
@@ -84,7 +84,19 @@ def _on_notification_conn_termination(conn: object) -> None:
...
+                    "Received internal event",
+                    event_=event,
+                )
+            else:
+                logger.info(
+                    "Received internal event",
+                    event_type=event.type,
+                    event_id=event.event_id.hex,
+                    organization=event.organization_id.str,
+                )
```
Comment (vxgmichel) [inline server/parsec/components/postgresql/events.py:98]: Use `info_with_debug_extra` here too

File: server/parsec/components/memory/factory.py [hunk 6]
```diff
@@ -32,7 +32,7 @@
 async def components_factory(config: BackendConfig) -> AsyncGenerator[dict[str, Any], None]:
     data = MemoryDatamodel({} if config.backend_mocked_data is None else config.backend_mocked_data)
 
-    async with event_bus_factory() as event_bus:
+    async with event_bus_factory(log_level=config.log_level) as event_bus:
```
Comment (vxgmichel) [inline server/parsec/components/memory/factory.py:35]: I would remove this as well

**Review by vxgmichel — CHANGES_REQUESTED** [review]

File: server/parsec/components/memory/factory.py (hunk 6 above)
Comment (touilleMan) [reply server/parsec/components/memory/factory.py:35]: :+1:

File: server/parsec/components/postgresql/events.py (hunk 5 above)
Comment (touilleMan) [reply server/parsec/components/postgresql/events.py:98]: :+1:

File: server/parsec/config.py (hunk 4 above)
Comment (touilleMan) [reply server/parsec/config.py:130]: :+1:

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:

**Pull request merged**